core/model/src/market.rs

```diff
@@ -41,6 +42,33 @@ impl RpcMessage for GetAgreement {
     type Error = RpcMessageError;
 }
 
+/// Lists all agreements
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ListAgreements {
+    pub state: Option<State>,
+    pub before_date: Option<DateTime<Utc>>,
+    pub after_date: Option<DateTime<Utc>>,
+    pub app_session_id: Option<String>,
+}
+
+impl Default for ListAgreements {
+    fn default() -> Self {
+        ListAgreements {
+            state: None,
+            before_date: None,
+            after_date: None,
+            app_session_id: None,
+        }
+    }
+}
```

Comment:
```suggestion

```

Reply:
use Default derive macro and delete implementation

Reply:
Done.